frontend/dockerfile/dockerfile_lint_test.go

```diff
@@ -41,8 +41,82 @@ var lintTests = integration.TestFuncs(
 	testBaseImagePlatformMismatch,
 	testAllTargetUnmarshal,
 	testRedundantTargetPlatform,
+	testSecretsUsedInArgOrEnv,
 )
 
+func testSecretsUsedInArgOrEnv(t *testing.T, sb integration.Sandbox) {
+	dockerfile := []byte(`
+FROM scratch
+ARG SECRET_PASSPHRASE
+ENV SUPER_Secret=foo
+ENV password=bar secret=baz
+ARG super_duper_secret_token=foo auth=bar
+ENV apikey=bar sunflower=foo
+ENV git_key=
+`)
+	checkLinterWarnings(t, sb, &lintTestParams{
+		Dockerfile: dockerfile,
+		Warnings: []expectedLintWarning{
+			{
+				RuleName:    "SecretsUsedInArgOrEnv",
+				Description: "Secrets should not be set as ARG or ENV",
```

Comment:
As we can't be 100% sure it is actually using secret value the message should be smth like "Potentially secret value set as ARG or ENV"